### ISSUE TITLE
Extend `identity_element` to more manifolds

### ISSUE THREAD

**Issue by mateuszbaran** [JuliaManifolds/Manifolds.jl#487]
See https://github.com/JuliaRobotics/IncrementalInference.jl/issues/1527 for discussion.

**Comment by kellertuer** [comment]
Isn't this already resolved now with #503 ?

**Comment by kellertuer** [comment]
AH, ok, then I mixed that up.

**Comment by kellertuer** [comment]
But still, to be precise I would only like to add `identity_element` to Lie groups to be precise?

**Comment by mateuszbaran** [comment]
I think the idea would be to also define `identity_element` to manifolds that can have a group structure but I'm not sure it's a good idea anyway.

**Comment by kellertuer** [comment]
I see that point, but I would prefer to be a little more precise. For example if the have `GL(n)` which has as manifold `Euclidean(n,n)` (not all of that, ok, I am not creative enough just now for a better example) what is the right identity? For `GL(n)` it would ne matrix multiplication so `I` but for the translation group on `Euclidean(n,n)` its `zeros(n,n)`. So just the manifold might not be enough to know the group structure and hence the identity element?

**Comment by mateuszbaran** [comment]
I don't think we currently have multiple groups structures with different identity elements (Euclidean and GL are topologically different) but in the future we could encounter such case, right.

**Comment by kellertuer** [comment]
Yes (that's why I was lazy with an example).

**Comment by kellertuer** [comment]
Coming back to this (looking through our issues moving a few to Lie groups) – I think an `identity_element` for manifolds does not make sense – and I now have the example I was missing a few years back

If we consider for some number `n` the two manifolds `M1 = Euclidean(n)` and `M2 = Euclidean(n,n)`

Then for `M1` the Lie group with the `AdditionGroupOperation` makes sense, for M2 as well.
If one is not careful enough `M2` with the MatrixMultiplicationGroupOperation can be used but would not make so much sense (we have `InvertibleMatrices` for a reason), so then

Should `identity_element(M1)` and `identity_element(M2)` be the zero vector? And only for `InvertibleMatrices` it is the unit matrix?

I personally would prefer to discontinue the `identity_element` in `Manifolds.jl` (when we remove the other group related functions) and just keep it for LieGroups, since there it is really clear. We would anyways for a manifold implicitly assume a group operation with respect to which we build the identity element. Maybe a bit implicit.

**Comment by kellertuer** [comment]
Yeah but both seem to be more like an utility and not precise enough (in my opinion) so having them in a utility package would be fine with me.
I was mainly going through a few old issues checking what would need to be done and whether to move them to LieGroups.

If we have a precise definition of both affine_matrix and identity_element (probably different names) what they might mean for a manifold, sure we can also add them here.
